bundler/lib/bundler/cli/gem.rb

```diff
@@ -238,13 +248,45 @@ def ask_and_set_test_framework
       test_framework
     end
 
-    def test_framework_hint
-      if Bundler.settings["gem.test"] == false
+    def hint_text(setting)
+      if Bundler.settings["gem.#{setting}"] == false
         "Your choice will only be applied to this gem."
       else
         "Future `bundle gem` calls will use your choice. " \
-        "This setting can be changed anytime with `bundle config gem.test`."
+        "This setting can be changed anytime with `bundle config gem.#{setting}`."
+      end
+    end
+
+    def ask_and_set_ci
+      ci_template = options[:ci] || Bundler.settings["gem.ci"]
+
+      if ci_template.to_s.empty?
+        Bundler.ui.confirm "Do you want to set up continuous integration for your gem? " \
+          "Supported services:\n" \
+          "* CircleCI:       https://circleci.com/\n" \
+          "* GitHub Actions: https://github.com/features/actions\n" \
+          "* GitLab CI:      https://docs.gitlab.com/ee/ci/\n" \
+          "* Travis CI:      https://travis-ci.org/\n" \
+          "\n"
+        Bundler.ui.info hint_text("ci")
+
+        result = Bundler.ui.ask "Enter a CI service. github/travis/gitlab/circle/(none):"
+        if result =~ /github|travis|gitlab|circle/
+          ci_template = result
+        else
+          ci_template = false
+        end
+      end
+
+      if Bundler.settings["gem.ci"].nil?
+        Bundler.settings.set_global("gem.ci", ci_template)
+      end
+
+      if options[:ci] == Bundler.settings["gem.ci"]
+        Bundler.ui.info "#{options[:ci]} is already configured, ignoring --ci flag."
```

Comment:
This is cute, I like it! :+1: Does it make sense to make this consistent with the `--test` flag?